src/main/java/edu/harvard/iq/dataverse/api/Admin.java

```diff
@@ -310,6 +312,37 @@ public Response deleteAuthenticatedUser(@PathParam("identifier") String identifi
 		}
 		return error(Response.Status.BAD_REQUEST, "User " + identifier + " not found.");
 	}
+        
+        @PUT
+        @Path("authenticatedUsers/changeIdentifier/{identifier}")
+        public Response changeAuthenticatedUserIdentifier(@PathParam("identifier") String oldIdentifier, String newIdentifier) {
+
+            if(null == oldIdentifier || oldIdentifier.isEmpty()) {
+                return error(Response.Status.BAD_REQUEST, "Old identifier provided to change is empty.");
+            } else if(null == newIdentifier || newIdentifier.isEmpty()) {
+                return error(Response.Status.BAD_REQUEST, "New identifier provided to change is empty.");
+            }
+            
+            AuthenticatedUser authenticatedUser = authSvc.getAuthenticatedUser(oldIdentifier);
+            if (authenticatedUser == null) {
+                return error(Response.Status.BAD_REQUEST, "User " + oldIdentifier + " not found in AuthenticatedUser");
+            }
+            
+            BuiltinUser builtinUser = builtinUserService.findByUserName(authenticatedUser.getUserIdentifier());
+            if (builtinUser == null) {
+                return error(Response.Status.BAD_REQUEST, "User " + oldIdentifier + " not found in BuiltinUser");
+            }
+            
+            List<RoleAssignment> raList = roleAssigneeSvc.getAssignmentsFor("@" + oldIdentifier); //only AuthenticatedUser supported
```

Comment:
would it make sense to put this directly in the command? (rather than call before and send as a param), same with builtinuser if we decide we don't need to validate it exists...

Reply:
I went with this route because the commands don't have access to the service EJB's (tho maybe I misunderstood)

Reply:
commands have access to ejbs via command context